src/lib/core/common-behaviors/common-module.ts

```diff
@@ -104,6 +107,18 @@ export class MatCommonModule {
     this._document.body.removeChild(testElement);
   }
 
+  /** Checks whether the material version matches the cdk version */
+  private _checkCdkVersionMatch(): void {
+    if (VERSION !== CDK_VERSION) {
```

Comment:
This will always evaluate to `false`, because the `VERSION` is an object. Also we probably want to check that only the major versions match (and potentially the minor ones?).

Reply:
given that the `package.json` specifies an exact version for the `@angular/cdk` peer dependency, I think comparing minor or even patch versions is useful.
https://github.com/angular/material2-builds/blob/1da8aff1873211de698c5d5d1e3e92e3530b5238/package.json#L26